eclipse-che/che-server
# fix: adding serviceAccounts imagePullSecrets into pod

Signed-off-by: xbaran4 <pbaran@redhat.com>



### What does this PR do?
Before creating a pod, adds ImagePullSecrets from ServiceAccount, that Pod uses, into the Pod.

Initial idea was to add ImagePullSecrets of a Pod (those that are added when user sets credentials for private repository) into the serviceAccount of the Pod. k8s/openshift would then mount them itself into the Pod.
This was however much more complicated (since after deletion of those secrets, serviceAccount should also delete them from ImagePullSecrets) and the new solution seems farily easy.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

fixes: https://github.com/eclipse/che/issues/19999

### How to test this PR?

Using che deployed on openshift with image quay.io/pbaran/che-server:pullsecrets:
1. Set any private registry credentials for your user in Che at user preferences tab.
2. Tag any image with openshift image registry public url (`oc regisry info --public=true`) and your user namespace and push it
3. In che, create a workspace using this image but with internal url (`oc regisry info --internal=true`)

The pod should successfully pull the image from openshift registry.

script for ease of testing: https://gist.github.com/xbaran4/41f3ef923e47fa74dd4bb03e8b625080

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
